core/src/main/java/discord4j/core/object/monetization/SKU.java

```diff
@@ -0,0 +1,214 @@
+package discord4j.core.object.monetization;
+
+import discord4j.common.util.Snowflake;
+import discord4j.core.GatewayDiscordClient;
+import discord4j.core.object.entity.Entity;
+import discord4j.discordjson.json.SkuData;
+
+import java.util.EnumSet;
+
+/**
+ * A Discord SKU.
+ *
+ * @see <a href="https://discord.com/developers/docs/monetization/skus">Discord Documentation</a>
+ */
+public class SKU implements Entity {
+
+    private final GatewayDiscordClient gateway;
+    private final SkuData data;
+
+    /**
+     * Constructs a {@code SKU} with an associated {@link GatewayDiscordClient} and Discord data.
+     *
+     * @param gateway The {@link GatewayDiscordClient} associated to this object.
+     * @param data The raw data as represented by Discord.
+     */
+    public SKU(final GatewayDiscordClient gateway, final SkuData data) {
+        this.gateway = gateway;
+        this.data = data;
+    }
+
+    /**
+     * Gets the type of SKU.
+     *
+     * @return The type of SKU.
+     */
+    public Type getType() {
+        return Type.of(data.type());
+    }
+
+    /**
+     * Gets the name of the SKU.
+     *
+     * @return The name of the SKU.
+     */
+    public String getName() {
+        return data.name();
+    }
+
+    /**
+     * Gets the application ID associated with this SKU.
+     *
+     * @return The application ID associated with this SKU.
+     */
+    public Snowflake getApplicationId() {
+        return Snowflake.of(data.applicationId());
+    }
+
+    /**
+     * Gets the SKU's slug.
+     *
+     * @return The SKU's slug.
+     */
+    public String getSlug() {
+        return data.slug();
+    }
+
+    /**
+     * Gets the flags of the SKU.
+     *
+     * @return An {@link EnumSet} of flags.
+     */
+    public EnumSet<Flag> getFlags() {
+        if (data.flags() != 0) {
+            return Flag.of(data.flags());
+        }
+        return EnumSet.noneOf(Flag.class);
+    }
+
+    /**
+     * Gets the data of the SKU.
+     *
+     * @return The data of the SKU.
+     */
+    public SkuData getSkuData() {
+        return data;
+    }
+
+    @Override
+    public GatewayDiscordClient getClient() {
+        return gateway;
+    }
+
+    @Override
+    public Snowflake getId() {
+        return Snowflake.of(data.id());
+    }
+
+    public enum Type {
```

Comment:
Missing DURABLE and CONSUMABLE type
ref: https://discord.com/developers/docs/monetization/skus#sku-object-sku-types